File: leetcode_ly/java/Array/RotateArray.java
public class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k = k % n;
        if(k == 0) return;
        int[] temp = new int[k];
        for(int i = 0; i < k; i++){
            temp[i] = nums[n - k + i];
        }
        for(int i = n - k - 1; i >= 0; i--){
            nums[i + k] = nums[i];
        }
        for(int i = 0; i < k; i++){
            nums[i] = temp[i];
        }
        return;
    }
}

File: leetcode_ly/c++/Array/RotateArray.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int len = nums.size();
        if(k <= 0){
            return;
        }
        else{
            k = k % len;
            nums.resize(len + len - k);
            for(int i = len; i < nums.size(); i++){
                nums[i] = nums[i - len];
            }
            for(int j = 0; j < len; j++){
                nums[j] = nums[j + len - k];
            }
            nums.resize(len);   
        }
    }
};